docs/reference/edot-sdks/java/setup/runtime-attach.md

```diff
@@ -0,0 +1,91 @@
+---
+navigation_title: Runtime attach Setup
+description: Guide on instrumenting Java applications using EDOT Java runtime attach.
+applies_to:
+  stack:
+  serverless:
+    observability:
+products:
+  - id: cloud-serverless
+  - id: observability
+  - id: edot-sdk
+---
+
+# Instrumenting Java applications with EDOT Java runtime attach
+
+% TODO do we have a global definition of what "tech preview" means that we could link to ?
+
+:::{warning}
+This feature is currently in *tech preview*.
+:::
+
+Runtime attach allows including the EDOT instrumentation agent in the application binary, doing so provides some setup [features](#features)
+and also comes with [limitations](#limitations) that must be reviewed beforehand.
+
+Adding runtime attach to an application is a 3-step process:
+- [add runtime attach to the application dependencies](#adding-runtime-attach-to-application-dependencies)
+- [minor code modification of the application main method](#minor-code-modification)
+- package and re-deploy the application
+
+## Features
+
+- Allows deploying the agent when access to JVM arguments or configuration is not possible, for example, with some managed services.
+- Agent deployment and update cycle can be controlled by the application development team, without needing modifications to the run environment.
+
+## Limitations
+
+- It can't be used with multiple applications that share a JVM, for example, with web-applications and application servers, in this case only the `-javaagent` setup option is supported
+- Agent update is tied to the application development and deployment lifecycle, it is not possible to update application and agent independently.
+- It requires minor modification of the application main entry point and adding one extra dependency.
+- Agent can only be attached at application start, it can't be used to attach later during application runtime.
+- Recent JVMs issue [warnings in standard error](#jvm-runtime-attach-warnings) and the feature might require explicit opt-in with JVM settings in the future
+ 
+## Adding runtime attach to application dependencies
+
+% TODO: can we replace ${VERSION} with the actual latest published version for an easier copy/paste experience ? 
```

Comment:
[for reviewer] This seems supported (https://docs-v3-preview.elastic.dev/elastic/docs-builder/tree/main/syntax/substitutions), but I don't know how to use it

Reply:
Yeah, I need to restore this automation!

Reply:
Just merged in `main`. Could you replace the version with `{{edot-java-version}}`?